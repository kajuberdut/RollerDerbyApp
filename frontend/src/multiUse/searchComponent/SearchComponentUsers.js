import './SearchComponentUsers.css'
import React, { startTransition, useState, useEffect } from 'react'
import {
Form
} from "reactstrap";
import SearchBarUsers from '../searchBar/SearchBarUsers';
import FastApi from '../../Api';

/**
 * Display search component 
 */

function SearchComponentUsers({getUsers, setUsers}) {

    /** Set initial state, set for data as initial state*/
  
    const INITIAL_STATE = {city: "", state: "", username: ""};
    const [formData, setFormData] = useState(INITIAL_STATE);


    /** Handle submit, get API result for companies or jobs, set formData to initial state*/
    
    const handleSubmit = evt => {
        evt.preventDefault();

        for (const key in formData) {
            if (formData[key] === null || formData[key] === "") {
            delete formData[key];
            }
        }
        
        // *note setting form data is taking time so I am using useEffect to run when formData has been changed 
        getData(formData)

        console.log(" $$$$$$$$$$ formData $$$$$$$$$$$$$:", formData)
        setFormData(INITIAL_STATE);
    };


    async function getData(formData) {
        console.log("formData ^^^^^^^^^^^^^^^^^^^", formData)
        let users = await FastApi.getUsers(formData)
        setUsers(users)
        console.log("users:", users)
    } 

    /** Render search component */

    return (
        <div className='SearchComponent'>
            <Form className="SearchComponentForm" onSubmit={handleSubmit}>
                <div className='SearchComponent-SB'>
                    <SearchBarUsers  formData={formData} setFormData={setFormData} />
                </div>
                <div className='SearchComponent-Btn'>
                    <button className='SearchComponentButton'>Search</button>
                </div>
            </Form>
        </div>
    );
  }
  
  export default SearchComponentUsers;