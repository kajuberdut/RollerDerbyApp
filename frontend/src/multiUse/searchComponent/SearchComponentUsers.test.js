import { render, screen } from '@testing-library/react';
import SearchComponentUsers from './SearchComponentUsers';

// * Passing as of 2/14/24
// npm test SearchComponentUsers.test.js
// must be in frontend directory 

test('renders SearchComponentUsers', () => {
  render(<SearchComponentUsers />);
});


test('matches snapshot', function() {
  const { asFragment } = render(<SearchComponentUsers />);
  expect(asFragment()).toMatchSnapshot(); 
})

