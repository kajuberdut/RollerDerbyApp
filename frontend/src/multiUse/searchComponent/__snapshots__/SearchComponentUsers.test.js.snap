// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="SearchComponent"
  >
    <form
      class="SearchComponentForm"
    >
      <div
        class="SearchComponent-SB"
      >
        <div
          class="SearchBarUsers"
        >
          <div
            class="SearchBarUsers-Username"
          >
            <input
              class="SearchBarInput form-control"
              id="searchbar-username"
              name="username"
              placeholder="Derby Name"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="SearchComponent-Btn"
      >
        <button
          class="SearchComponentButton"
        >
          Search
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
